### ISSUE TITLE
 执行finetune.sh遇到exits with return code = -9是怎么错误呢？

### ISSUE THREAD

**Issue by PapaMadeleine2022** [LlamaFamily/Llama-Chinese#364]
你好，我下载了Llama3-Chinese-8B-Instruct模型，在train/sft目录下执行 sh finetune.sh，解决了遇到的环境冲突问题后，终于要训起来的时候，报下面错误：

'''
...
[INFO|trainer.py:607] 2024-10-27 16:31:32,759 >> Using auto half precision backend
0 start train
[2024-10-27 16:31:33,323] [INFO] [logging.py:96:log_dist] [Rank 0] DeepSpeed info: version=0.14.0, git-hash=unknown, git-branch=unknown
[2024-10-27 16:31:33,754] [INFO] [logging.py:96:log_dist] [Rank 0] DeepSpeed Flops Profiler Enabled: False
Installed CUDA version 11.8 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Using /home/sankuai/.cache/torch_extensions/py39_cu117 as PyTorch extensions root...
Detected CUDA files, patching ldflags
Emitting ninja build file /home/sankuai/.cache/torch_extensions/py39_cu117/cpu_adam/build.ninja...
Building extension module cpu_adam...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: no work to do.
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.8009004592895508 seconds
Adam Optimizer #0 is created with AVX2 arithmetic capability.
Config: alpha=0.000100, betas=(0.900000, 0.999000), weight_decay=0.000000, adam_w=1
[2024-10-27 16:31:40,566] [INFO] [logging.py:96:log_dist] [Rank 0] Using DeepSpeed Optimizer param name adamw as basic optimizer
[2024-10-27 16:31:40,566] [INFO] [logging.py:96:log_dist] [Rank 0] Removing param_group that has no 'params' in the basic Optimizer
[2024-10-27 16:31:40,581] [INFO] [logging.py:96:log_dist] [Rank 0] DeepSpeed Basic Optimizer = DeepSpeedCPUAdam
[2024-10-27 16:31:40,581] [INFO] [utils.py:56:is_zero_supported_optimizer] Checking ZeRO support for optimizer=DeepSpeedCPUAdam type=<class 'deepspeed.ops.adam.cpu_adam.DeepSpeedCPUAdam'>
[2024-10-27 16:31:40,581] [INFO] [logging.py:96:log_dist] [Rank 0] Creating torch.bfloat16 ZeRO stage 2 optimizer
[2024-10-27 16:31:40,581] [INFO] [stage_1_and_2.py:149:__init__] Reduce bucket size 500000000
[2024-10-27 16:31:40,581] [INFO] [stage_1_and_2.py:150:__init__] Allgather bucket size 500000000
[2024-10-27 16:31:40,581] [INFO] [stage_1_and_2.py:151:__init__] CPU Offload: True
[2024-10-27 16:31:40,581] [INFO] [stage_1_and_2.py:152:__init__] Round robin gradient partitioning: False
[2024-10-27 16:32:12,343] [INFO] [utils.py:800:see_memory_usage] Before initializing optimizer states
[2024-10-27 16:32:12,344] [INFO] [utils.py:801:see_memory_usage] MA 16.06 GB         Max_MA 16.06 GB         CA 16.56 GB         Max_CA 17 GB 
[2024-10-27 16:32:12,344] [INFO] [utils.py:808:see_memory_usage] CPU Virtual Memory:  used = 35.33 GB, percent = 38.4%
[2024-10-27 16:32:48,277] [INFO] [launch.py:316:sigkill_handler] Killing subprocess 210479
[2024-10-27 16:32:48,284] [ERROR] [launch.py:322:sigkill_handler] ['/home/xxx/tools/anaconda3/envs/conda_py3.9/bin/python3.9', '-u', 'finetune_clm.py', '--local_rank=0', '--model_name_or_path', '../../../Llama3-Chinese-8B-Instruct', '--train_files', '../../data/train_sft.csv', '--validation_files', '../../data/dev_sft.csv', '../../data/dev_sft_sharegpt.csv', '--per_device_train_batch_size', '1', '--per_device_eval_batch_size', '1', '--do_train', '--do_eval', '--use_fast_tokenizer', 'false', '--output_dir', 'save_folder', '--evaluation_strategy', 'steps', '--max_eval_samples', '800', '--learning_rate', '1e-4', '--gradient_accumulation_steps', '8', '--num_train_epochs', '100', '--warmup_steps', '400', '--logging_dir', 'save_folder/logs', '--logging_strategy', 'steps', '--logging_steps', '10', '--save_strategy', 'steps', '--preprocessing_num_workers', '1', '--save_steps', '20', '--eval_steps', '20', '--save_total_limit', '2000', '--seed', '42', '--disable_tqdm', 'false', '--ddp_find_unused_parameters', 'false', '--block_size', '128', '--report_to', 'tensorboard', '--overwrite_output_dir', '--deepspeed', 'ds_config_zero2.json', '--ignore_data_skip', 'true', '--bf16', '--gradient_checkpointing', '--bf16_full_eval', '--ddp_timeout', '18000000'] exits with return code = -9
'''

这个log完全看不出是什么问题啊?  求指点！！

**Comment by PapaMadeleine2022** [comment]
使用一张A100训练的时候报上面的错误，使用2张时没有报上面错误了。
